### PR TITLE
[ENH] Add pandas_compat.table_to_frame(tab)

### PR DESCRIPTION
**Pull request by apetrov**
### Feature

Convert Orange.data.Table instance to pandas dataframe

##### Includes
- [X] Code changes
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Orange/data/pandas_compat.py [hunk 1]
```diff
@@ -76,3 +76,42 @@ def _is_datetime(s):
...
+    Parameters
+    ----------
+    tab : Table
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    def _column_to_series(col, vals):
+        print(col.name)
```
Comment (astaric) [inline Orange/data/pandas_compat.py]: I guess this was used for debugging. Please remove.

File: Orange/data/pandas_compat.py [hunk 2]
```diff
@@ -76,3 +76,42 @@ def _is_datetime(s):
...
+        if col.is_discrete:
+            labels = [col.values[i] for i in np.vectorize(int)(vals)]
```
Comment (kernc) [inline Orange/data/pandas_compat.py]: `vals.astype(int)` avoids a call into python.

But what if some values are NaN?

Alternative:
```py
return col.name, pd.Categorical.from_codes(codes=pd.Series(vals).fillna(-1).astype(int),
                                           categories=col.values, ordered=col.ordered)

File: Orange/data/pandas_compat.py [hunk 3]
```diff
@@ -76,3 +76,42 @@ def _is_datetime(s):
...
+            return (col.name, pd.Series(labels).astype('category'))
+        elif col.is_time:
+            return (col.name, pd.to_datetime(vals,unit='s').to_series())
```
Comment (kernc) [inline Orange/data/pandas_compat.py]: spaces after commas.

File: Orange/data/pandas_compat.py [hunk 4]
```diff
@@ -76,3 +76,42 @@ def _is_datetime(s):
...
+            return (col.name, pd.Series(vals))
+    def _columns_to_series(cols, vals):
+        return [_column_to_series(col, vals[:,i]) for i,col in enumerate(cols)]
+
+    x,y,metas = [],[],[]
+    domain = tab.domain
+    if domain.attributes:
+        x = _columns_to_series(domain.attributes,tab.X)
+    if domain.class_vars:
+        y = _columns_to_series(domain.class_vars,tab.Y.reshape(tab.Y.shape[0],1))
```
Comment (kernc) [inline Orange/data/pandas_compat.py]: Reshaping Y here doesn't work with multiple target columns. Perhaps:
```py
table.Y.reshape(table.Y.shape[0], len(domain.class_vars))

File: Orange/data/pandas_compat.py (hunk 1 above)
Comment (apetrov) [reply Orange/data/pandas_compat.py]: thanks

File: Orange/data/pandas_compat.py (hunk 2 above)
Comment (apetrov) [reply Orange/data/pandas_compat.py]: Great suggestion, thanks, fixed.

File: Orange/data/pandas_compat.py (hunk 4 above)
Comment (apetrov) [reply Orange/data/pandas_compat.py]: Thanks

**Comment by apetrov** [comment]
@astaric sure, added basic tests and looks good on Iris data set.
checking for datasets with datetime columns so I could test datetime columns properly.

**Comment by apetrov** [comment]
update: fixed pylinting.

**Comment by apetrov** [comment]
ok, looks like a couple of datasets doesn't work with it:
```python
from Orange.data import Table
from Orange.data.pandas_compat import table_to_frame
datasets = [ f.split('.')[0] for f in os.listdir("Orange/datasets") if '.tab' in f]
for ds in datasets:
    print(ds)
    table = Table(ds)
    table_to_frame(table)
~

**Comment by astaric** [comment]
I'd add it as a test, but mark it as @skip, so it can be uncommented when needed.

You wrote in an earlier comment that the function did not work on a couple of datasets. Could you test on them as well? (not sure what the problem was)

**Comment by apetrov** [comment]
@astaric ok, that's a good point.

Yes, it helped me to troubleshoot a problem with datetime column. works on any Orange build-in dataset

**Pull request merged**